### DISCUSSION TITLE
Unable to find rule for use case, no-implicit-coercion missing rule?

### DISCUSSION THREAD

**Discussion by jockjocko** [eslint/eslint#17041] [Q&A] [answered]
I'm trying to prevent type coercion of a object to a number when using the Subtract operator as it resulted in a hard to catch bug as the return type of the coercion is NaN which in typescript is of the type number.

```typescript
// const balance = 0; was of type number before it was changed

// Bug introduced by changing to an object. type checking succeeds as expected but NaN was not a desired value.

const balance = { amount: 0, currency: 'GBP' };

// Why is this coercion not flagged as a warning by no-implicit-coercion
const amount: number = -balance;

// This is correctly flagged as a warning by no-implicit-coercion
const amount: number = +balance;

console.log(negativeAmount); // NaN 

console.log(positiveAmount); // NaN

// Example function that would always return NaN due to not checking for NaN
const totalUpAllBalances = (amount: number) =>  {
  const total = 0;
  total += amount
}


```

The rule no-implicit-coercion correctly catches the + operator but not the - operator
Is there another rule that would prevent this from happening or is the no-implicit-coercion missing the - operator?

**Comment by mdjermanovic** [comment, marked as answer]
The `no-implicit-coercion` rule doesn't have type information.

`+balance` is reported because if `balance` is a number then`+` does nothing, if `balance` is not a number then `+` converts it to a number, so the rule assumes that `+balance` is an implicit coercion.

With `-balance`, we cannot know if the intent is to negate a number or convert another type to a number.

**Reply by jockjocko** [reply]
Thanks for the explanation. 👍 this might be one for the typescript team to look at.
